### PR TITLE
REF: Use NativeStack

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/zzh9maf214ggd7rgj5c8er4vx0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8194)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/y2qed133579qzv5cnxxy351kbg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8195)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/v95ev0vn9y44tz616yt211vf54?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8196)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/n6bukgv8zynaaq3nq3peqy82h0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8197)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xdcm8ffbp1uazuh5dwtk29d4m0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8198)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/udjqyuvkutmh31jbwv8a0pm9d4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8206)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ztcp6mfbw8rbw7nwv74p2m7qdr?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8209)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/u0n4ga3uvw1behg3aatumw19tm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8212)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/pkf0ehqfvhh268zw4jv7bq3jx8?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8213)

**Comment by marcosrdz** [comment]
Greatly improves table and macOS UX. Performance improvement for navigation on iOS and Android. Has what's needed for continuing Windows work. 

@ncoelho  please check for any styling issues

Currently looking at why e2e fails even though I am able to manually succeed the flow

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/tj6wrt58212jd7uw8nyktz03yg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8215)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/e38hbjcut14z8cxkxj5n8m0d60?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8217)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/yk6f90ceu0dfv747mf643kqghg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8221)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/4kuf6ah5mm5y5pd3ptvheptcvg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8222)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/b2xjz2txdtyh6xn5ve8a3zdwpc?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8223)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ncw8cqt3t5xwfvkmd6zfz9wger?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8224)

**Review by limpbrains — APPROVED** [review]

**Review by ncoelho — CHANGES_REQUESTED** [review]
- [ ]  Lightning > refill with external wallet not loading. And freezes the app, after this error go to onchain wallet and send screen doesn't load.

- [ ] Confirmations element lost the style

E-SV3w0UUAE5kvv

- [ ] denomination lost the style (color) on send screen 
Screenshot 2021-09-10 at 09 49 38

- [ ] There's a "color glitch" on moving from the list to the transaction screen on the navigation header.

https://user-images.githubusercontent.com/1238429/132819858-50aefa78-7dd7-41a1-8790-34a4b294dd6f.mov

- [ ] Status bar goes white on settings screen on android.

- [ ] When going from the wallet view to the main view, there's an abrupt change on the status bar color (light mode), I've seen other react apps to this change smoothly with a fade animation instead of an abrupt change. Details, but would be nice to see if it can be improved.

- [ ] there's a close (x) on the create wallet backup screen.

- [ ] there's a jump on initializing the app where the navigation bar loads only after everything.

- [ ] when going to the settings screen, it loads with the big title hidden :) only after scrolling it appears.

- [ ] Reoder screen header, looks weird. Something missing there, is it push or pull?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/uqeev5ydurzakp67ewwvq0tdw0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8229)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: screen/wallets/addMultisigStep2.js [hunk 1]
```diff
@@ -145,35 +144,37 @@ const WalletsAddMultisigStep2 = () => {
...
+  //             // If the user confirmed, then we dispatch the action we blocked earlier
+  //             // This will continue the action that had triggered the removal of the screen
+  //             onPress: () => navigation.dispatch(e.data.action),
+  //           },
+  //         ]);
+  //       }
+  //     }
+  //   });
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [navigation, hasUnsavedChanges, cosigners]);
```
Comment (Overtorment) [inline screen/wallets/addMultisigStep2.js:177]: dead code. usually, its better to remove it.

File: screen/wallets/hodlHodl.js [hunk 2]
```diff
@@ -150,15 +150,17 @@ export default class HodlHodl extends Component {
     this.setState({ methods });
   }
 
-  onFocus = async () => {
-    const hodlApiKey = await this.context.getHodlHodlApiKey();
-    const displayLoginButton = !hodlApiKey;
+  onFocus = async e => {
+    if (e.data.closing === false) {
```
Comment (Overtorment) [inline screen/wallets/hodlHodl.js:154]: return early, and you will be able to revert to previous indentation, and less LOC will be marked as changed

File: screen/wallets/reorderWallets.android.js [hunk 3]
```diff
@@ -88,20 +88,30 @@ const ReorderWallets = () => {
...
+          onPress={() => {
+            if (sortableList.current?.state.data.length === data.length && hasMovedARow) {
+              const newWalletsOrderArray = [];
+              sortableList.current.state.order.forEach(element => {
+                newWalletsOrderArray.push(data[element]);
+              });
+              setWalletsWithNewOrder(newWalletsOrderArray);
+            }
+            navigation.goBack();
+          }}
```
Comment (Overtorment) [inline screen/wallets/reorderWallets.android.js:105]: it appears that reordering code is duplicated in android & ios.
its better to isolate it in a separate func. you could even write tests for it.
also, if you're worried about it, you could add some safety checks in `setWalletsWithNewOrder()`, e.g. old array length equals new array length, and its greater than 0

File: screen/wallets/viewEditMultisigCosigners.js [hunk 4]
```diff
@@ -135,33 +135,29 @@ const ViewEditMultisigCosigners = () => {
...
+  //           style: 'destructive',
+  //           // If the user confirmed, then we dispatch the action we blocked earlier
+  //           // This will continue the action that had triggered the removal of the screen
+  //           onPress: () => dispatch(e.data.action),
+  //         },
+  //       ]);
+  //     }
+  //   });
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [isSaveButtonDisabled]);
```
Comment (Overtorment) [inline screen/wallets/viewEditMultisigCosigners.js:160]: dead code

**Review by Overtorment — CHANGES_REQUESTED** [review]

**Comment by Overtorment** [comment]
i also tested a bit on appetize and haven't found any issues

File: screen/send/psbtMultisig.js [hunk 5]
```diff
@@ -157,8 +157,8 @@ const PsbtMultisig = () => {
   }, [receivedPSBTBase64]);
 
   const _combinePSBT = () => {
-    const receivedPSBT = bitcoin.Psbt.fromBase64(receivedPSBTBase64);
     try {
+      const receivedPSBT = bitcoin.Psbt.fromBase64(receivedPSBTBase64);
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply screen/send/psbtMultisig.js:161]: It was outside the try catch so if invalid data was fed it would crash the app instead of throwing an alert

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/advqygqbg3m13x8tqtedq329hg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8237)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/yy4wcd473efdkbvjg9huak5vag?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8244)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/w49k97cmbkjzwpr84vv9n85ub0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8245)

File: screen/wallets/reorderWallets.android.js (hunk 3 above)
Comment (marcosrdz) [reply screen/wallets/reorderWallets.android.js:105]: It has different renders. I would wait for an updated package with a fix to merge em into 1 file. The android renders must be different from the Mac and iOS renders

Comment (marcosrdz) [reply screen/wallets/reorderWallets.android.js:105]: Not only different renders but also different functions for when rows get moved

File: screen/wallets/addMultisigStep2.js [hunk 6]
```diff
@@ -68,8 +67,8 @@ const WalletsAddMultisigStep2 = () => {
   const [importText, setImportText] = useState('');
   const openScannerButton = useRef();
   const data = useRef(new Array(n));
-  const hasUnsavedChanges = Boolean(cosigners.length > 0 && cosigners.length !== n);
-  const isDiscardConfirmAlertPresented = useRef(false);
+  // const hasUnsavedChanges = Boolean(cosigners.length > 0 && cosigners.length !== n);
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply screen/wallets/addMultisigStep2.js:70]: Shall it be removed or kept commented for once the package supports it?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/4znnar3rdhxm4uf1wv3x754aw4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8246)

**Comment by marcosrdz** [comment]
> * [x ]   Lightning > refill with external wallet not loading. And freezes the app, after this error go to onchain wallet and send screen doesn't load.
> * [ ]  Confirmations element lost the style
> 
> E-SV3w0UUAE5kvv
> 
> * [x ]  denomination lost the style (color) on send screen
> 
> Screenshot 2021-09-10 at 09 49 38
> 
> * [ ]  There's a "color glitch" on moving from the list to the transaction screen on the navigation header.
> 
>  Screen.Recording.2021-09-10.at.09.52.01.mov 
> 
> * [ ]  Status bar goes white on settings screen on android.
> * [ ]  When going from the wallet view to the main view, there's an abrupt change on the status bar color (light mode), I've seen other react apps to this change smoothly with a fade animation instead of an abrupt change. Details, but would be nice to see if it can be improved.
> * [x ]  there's a close (x) on the create wallet backup screen.
> * [ ]  there's a jump on initializing the app where the navigation bar loads only after everything.

https://github.com/software-mansion/react-native-screens/pull/1029



> * [x ]  when going to the settings screen, it loads with the big title hidden :) only after scrolling it appears.
> * [ ]  Reoder screen header, looks weird. Something missing there, is it push or pull?
its a fullscreen modal so gestures don't hijack wallet reordering

> * [ x]  lightning invoice screen has both back and close icons.
> * [ ]  transaction success screen, success animation should be centered.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/m75a32th9agtk3g9evbtyk1wxc?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8247)

File: screen/settings/settings.js [hunk 7]
```diff
@@ -22,7 +22,7 @@ const Settings = () => {
   return (
     <ScrollView style={styles.root}>
       <StatusBar barStyle="default" />
-      <BlueHeaderDefaultSub leftText={loc.settings.header} />
+      {Platform.OS === 'android' && <BlueHeaderDefaultSub leftText={loc.settings.header} />}
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply screen/settings/settings.js]: Ill add it but im guessing are referring to Viewcomponents on Flatlists

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/zzze8z59q8nghvj432yjcv45j0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8249)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/t14x81bnfh29cmgb6bmfn98guc?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8248)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/a1w4jj2k5wktqb26a1521fufkm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8251)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/nn8q6j18rnvv8fhg1j92p22uy4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8252)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/t9jv621qzg7dyh0tp9t7n7dz2c?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8256)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/868kdb23ympzkk3z2455r64bt0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8257)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/fz1c4t49ddy9tzhn940019etzw?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8264)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/k662r87gnayt65m63q5429ny3w?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8266)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @ncoelho] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/ncoelho

**Comment by Overtorment** [comment]
because this pr has screens bump as well, travis fails.
is it possible to have this pr but without screens bump?

**Comment by marcosrdz** [comment]
> because this pr has screens bump as well, travis fails.
> is it possible to have this pr but without screens bump?

Just lowered it to the one currently in use. e2e should pass

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/tj7t2nujzqqj309242kfdkjx48?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8277)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/mxhqv6gmp08j49tcn8ezjfkwtg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8276)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/0c0z1vuf7efmt7y00y3bpgqxar?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8279)

**Review by ncoelho — APPROVED** [review]

**Pull request merged**